src/map/skill.cpp

```diff
@@ -23017,6 +23058,8 @@ void skill_reload (void) {
 	abra_db.clear();
 	magic_mushroom_db.clear();
 	reading_spellbook_db.clear();
+	skill_arrow_db.clear();
+
 	skill_readdb();
 	initChangeTables(); // Re-init Status Change tables
 
```

Comment:
Please add the missing calls to do_final_skill (also applies to other dbs).
We seem to have missed it so far.